spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/RepositorySchemaController.java

```diff
@@ -31,24 +34,38 @@
  * 
  * @author Jon Brisbin
  * @author Oliver Gierke
+ * @author Greg Turnquist
+ * @see http://json-schema.org/
  */
-@RepositoryRestController
+@BasePathAwareController
 class RepositorySchemaController {
 
-	private static final String BASE_MAPPING = "/{repository}";
-
 	private final PersistentEntityToJsonSchemaConverter jsonSchemaConverter;
+	private final RepositoryResourceMappings mappings;
```

Comment:
These fields don't seem to be used anywhere? Can the constructor be cleaned up, too?